app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java

```diff
@@ -569,12 +574,84 @@ private VideoOptions buildExoPlayerOptions(@Nullable Integer forcedSubtitleIndex
         DeviceProfile internalProfile = new ExoPlayerProfile(
                 mFragment.getContext(),
                 isLiveTv && !userPreferences.getValue().get(UserPreferences.Companion.getLiveTvDirectPlayEnabled()),
-                userPreferences.getValue().get(UserPreferences.Companion.getAc3Enabled())
+                getDirectPlayPreferences(),
+                getAudioTranscodeTarget(),
+                userPreferences.getValue().get(UserPreferences.Companion.getEnable4kSupport())
         );
         internalOptions.setProfile(internalProfile);
         return internalOptions;
     }
 
+    @NonNull
+    private String[] getDirectPlayPreferences() {
+        ArrayList<String> codecs = new ArrayList<>();
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getMpegEnabled())) {
+            codecs.add(Codec.Audio.MP2);
+            codecs.add(Codec.Audio.MP3);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getAacEnabled())) {
+            codecs.add(Codec.Audio.AAC);
+            codecs.add(Codec.Audio.AAC_LATM);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getAc3Enabled())) {
+            codecs.add(Codec.Audio.AC3);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getEac3Enabled())) {
+            codecs.add(Codec.Audio.EAC3);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getDtsEnabled())) {
+            codecs.add(Codec.Audio.DTS);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getThdEnabled())) {
+            codecs.add(Codec.Audio.TRUEHD);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getPcmEnabled())) {
+            codecs.add(Codec.Audio.PCM_ALAW);
+            codecs.add(Codec.Audio.PCM_MULAW);
+        }
+
+        if(userPreferences.getValue().get(UserPreferences.Companion.getOtherAudioEnabled())) {
+            codecs.add(Codec.Audio.DCA);
+            codecs.add(Codec.Audio.MLP);
+            codecs.add(Codec.Audio.OPUS);
+            codecs.add(Codec.Audio.FLAC);
```

Comment:
Since you’re adding a new panel and there are only two other codecs, I would add them as separate options.